compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -1544,19 +1747,19 @@ fn var_name(tcx: TyCtxt<'_>, var_hir_id: hir::HirId) -> Symbol {
     tcx.hir().name(var_hir_id)
 }
 
-fn should_do_migration_analysis(tcx: TyCtxt<'_>, closure_id: hir::HirId) -> bool {
-    let (level, _) =
-        tcx.lint_level_at_node(lint::builtin::DISJOINT_CAPTURE_DROP_REORDER, closure_id);
+fn should_do_disjoint_capture_migration_analysis(tcx: TyCtxt<'_>, closure_id: hir::HirId) -> bool {
+    let (level_drop, _) =
+        tcx.lint_level_at_node(lint::builtin::DISJOINT_CAPTURE_MIGRATION, closure_id);
 
-    !matches!(level, lint::Level::Allow)
+    !matches!(level_drop, lint::Level::Allow)
 }
 
 /// Return a two string tuple (s1, s2)
 /// - s1: Line of code that is needed for the migration: eg: `let _ = (&x, ...)`.
 /// - s2: Comma separated names of the variables being migrated.
 fn migration_suggestion_for_2229(
     tcx: TyCtxt<'_>,
-    need_migrations: &Vec<hir::HirId>,
+    need_migrations: &FxHashSet<hir::HirId>,
```

Comment:
we should keep this a vector; we want to preserve order. otherwise things might get dropped in a different ordered. 

Though we have a ticket to fix it overall, so might not be too bad 🙃

Reply:
Nvm i see why you're using hashset. I'm dumb

Reply:
Tis ok, I figured out how to use vector instead without have dups